tchannel/testing/vcr/cassette.py

```diff
@@ -39,20 +39,55 @@
 VERSION = 1
 
 
+def primitive_struct(obj):
+    data = {}
+    for field in obj.type_spec.fields:
+        attr = getattr(obj, field.name)
+        if attr is not None:
+            data[field.name] = primitive(attr, field.ttype_code)
+
+    return data
+
+
+def primitive_list(obj):
+    # I think this is because our objects aren't hydrating with the correct
+    # subtypes.
+    if obj and isinstance(obj[0], (str, dict)):
+        return obj
+    return [primitive(v, v.type_spec.ttype_code) for v in obj]
+
+
+def primitive(value, ttype_code):
+    return ({
+        TType.LIST: lambda: primitive_list(value),
+        TType.STRUCT: lambda: primitive_struct(value),
+    }.get(ttype_code) or (lambda: value))()
+
+
 class Interaction(namedtuple('Interaction', 'request response')):
     """An interaction is a request-response pair."""
 
     def to_primitive(self):
+        # import ipdb; ipdb.set_trace()
+        # from thriftrw.protocol import BinaryProtocol
+        # p = BinaryProtocol()
+
+        if hasattr(self.request, 'to_primitive'):
+            return {
+                'request': self.request.to_primitive(),
+                'response': self.response.to_primitive(),
+            }
```

Comment:
I assume this is to work with the old code? You probably want to delete this now.